packages/manager/apps/pci-instances/src/data/hooks/instance/useInstances.ts

```diff
@@ -42,26 +46,32 @@ export const updateInstanceFromCache: TUpdateInstanceFromCache = (
   payload,
 ) => {
   const { projectId, instance } = payload;
-  if (!instance.id) return;
-  queryClient.setQueriesData<InfiniteData<TInstanceDto[], number>>(
-    {
-      predicate: listQueryKeyPredicate(projectId),
-    },
-    (prevData) => {
-      if (!prevData) return undefined;
-      const updatedPages = prevData.pages.map((page) =>
-        page.map((prevInstance) =>
-          instance.id === prevInstance.id
-            ? {
-                ...prevInstance,
-                ...instance,
-              }
-            : prevInstance,
-        ),
-      );
-      return { ...prevData, pages: updatedPages };
-    },
-  );
+
+  const queries = queryClient.getQueriesData<InfiniteData<TInstanceDto[]>>({
+    predicate: listQueryKeyPredicate(projectId),
+  });
+
+  queries.forEach(([queryKey, queryData]) => {
+    if (!queryData) return;
+
+    const updatedPages = queryData.pages.map((page) =>
+      page.map((prevInstance) =>
+        prevInstance.id === instance.id
+          ? { ...prevInstance, ...instance }
+          : prevInstance,
+      ),
+    );
+
+    if (isEqual(queryData.pages, updatedPages)) return;
```

Comment:
You could have a findIndex, if found you copy the array and replace the value, if not you return

This would avoid a lot of computations (double map and isEqual which does deep compare)

You could also have a boolean which handles if the condition was true at a time

Reply:
Yes, I added a deep compare at instance level, that is much more efficient, thx 👍🏻